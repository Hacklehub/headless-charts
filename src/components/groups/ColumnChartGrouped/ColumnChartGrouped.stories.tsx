import ColumnChartGrouped from '.';
import { Meta } from '@storybook/react';
import data from './sample.json';

export default {
  title: 'Groups/ColumnChartGrouped',
  component: ColumnChartGrouped,
  tags: ['autodocs'],
} as Meta;

export const Default = {
  args: {
    data,
    id: 'column-chart-group-default',
    x: { key: "name" },
    y: [
      {
        key: "USA",
        start: 0,
      },
      { key: "Europe" },
      { key: "APAC"},
      { key: "Africa"},
    ],
  },
};

export const Styled = {
  args: {
    data,
    id: 'column-chart-group-styled',
    className: 'bg-gray-100 rounded',
    padding: {
      top: 20,
      right: 20,
      bottom: 20,
      left: 5,
      bar: 0.1,
    },
    margin: {
      top: 0,
      right: 40,
      bottom: 40,
      left: 60,
    },
    x: [
      {
        key: 'reading',
        className: 'fill-red-500 rounded',
      },
      { key: 'value', className: 'fill-blue-500' },
    ],
    y: [
      {
        key: 'USA',
        start: 0,
        className: 'text-purple-300',
      },
      { key: 'Europe', className: 'text-purple-500 ' },
      { key: 'APAC', className: 'text-purple-700' },
      { key: 'Africa', className: 'text-purple-900' },
    ],
  },
};
