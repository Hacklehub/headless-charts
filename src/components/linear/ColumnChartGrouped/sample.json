[
  {
    "name": "Product A",
    "USA": 10000,
    "Europe": 4000,
    "APAC": 5000,
    "Africa": 1000
  },
  { "name": "Product B", "USA": 9400, "Europe": 9000, "APAC": 4000 },
  { "name": "Product C", "USA": 8000, "Europe": 4000, "APAC": 1000 }
]
