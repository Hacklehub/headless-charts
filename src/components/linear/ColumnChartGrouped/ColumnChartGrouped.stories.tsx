import React, { useState } from 'react';

/* eslint-disable @typescript-eslint/ban-ts-comment */
/* eslint-disable @typescript-eslint/no-explicit-any */
import {transition} from 'd3-transition';
import ColumnChartGrouped from '.';
import { Meta } from '@storybook/react';
import data from './sample.json';

export default {
  title: 'Linear/ColumnChartGrouped',
  component: ColumnChartGrouped,
  tags: ['autodocs'],
} as Meta;

export const Default = {
  args: {
    data,
    id: 'column-chart-group-default',
    x: { key: 'name' },
    y: [
      {
        key: 'USA',
        start: 0,
      },
      { key: 'Europe' },
      { key: 'APAC' },
      { key: 'Africa' },
    ],
  },
};

export const Styled = {
  args: {
    data,
    id: 'column-chart-group-styled',
    className: 'bg-gray-100 rounded',
    padding: {
      top: 20,
      right: 20,
      bottom: 20,
      left: 5,
      bar: 0.1,
    },
    margin: {
      top: 10,
      right: 40,
      bottom: 40,
      left: 60,
    },
    x: { key: 'name', className: 'fill-blue-500' },
    y: [
      {
        key: 'USA',
        start: 0,
        className: 'text-purple-300',
      },
      { key: 'Europe', className: 'text-purple-500 ' },
      { key: 'APAC', className: 'text-purple-700' },
      { key: 'Africa', className: 'text-purple-900' },
    ],
  },
};

export const Animated = {
  args: {
    ...Default.args,
    drawing: { duration: 1000 },
  },
};

export const WithTooltip = {
  args: {
    ...Default.args,
    tooltip: {
      className: 'bg-gray-100 rounded p-2',
    },
  },
};

export const WithCustomTooltip = {
  args: {
    ...Default.args,
    tooltip: {
      html: (data: any) => {
        return `
          <div class="bg-gray-100 rounded p-2">
            <div class="text-sm font-semibold">${data.name}</div>
            <div class="text-xs">${data['USA']}</div>
          </div>
        `;
      },
    },
  },
};

// export const DataRefreshWithDrawing = () => {
//   const [pieData, setPieData] = useState(data);
//   const refreshData = () => {
//     setPieData(
//       pieData.map((d) => ({ ...d, USA: d['USA'] + Math.random() * 1000 }))
//     );
//   };
//   return (
//     <div>
//       <button onClick={refreshData}>Refresh</button>
//       <ColumnChartGrouped id='column-chart-grouped' data={pieData} />
//     </div>
//   );
// };

// The above should be remove

export const UpdatingData = () => {
  const [columnChartData, setColumnChartData] = React.useState(data);
  const updateData = () => {
    setColumnChartData(
      columnChartData.map((d) => ({
        ...d,
        USA:  Math.random() * 1000,
        Europe: Math.random() * 1000,
        APAC: Math.random() * 1000,
      })))
  }

  const t = transition().duration(1000);
  console.log(columnChartData)
  return (
    <>
      <a onClick={()=>{
        updateData()
      }}> {"Update Data"} </a>
      <ColumnChartGrouped
        id='column-chart-grouped-updating-data'
        data={columnChartData}
        x={{
          key: 'name',
          axis: 'bottom',
        }}
        y={[
          {
            key: 'USA',
            start: 0,
            className: 'text-purple-300',
          },
          { key: 'Europe', className: 'text-purple-500 ' },
          { key: 'APAC', className: 'text-purple-700' },
          { key: 'Africa', className: 'text-purple-900' },
        ]}
        transition={t}
      />
    </>
  );
};
