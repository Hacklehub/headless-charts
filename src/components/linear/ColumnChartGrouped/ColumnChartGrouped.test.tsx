import { describe, expect, it } from 'vitest';
import { render, screen } from '@testing-library/react';

import ColumnChartGrouped from '.';

const data = [
  {
    name: 'Product A',
    USA: 10000,
    Europe: 4000,
    APAC: 5000,
    Africa: 1000,
  },
  { name: 'Product B', USA: 9400, Europe: 9000, APAC: 4000, Africa: 0 },
  { name: 'Product C', USA: 8000, Europe: 4000, APAC: 1000, Africa: 0 },
];

describe('ColumnChartGrouped', () => {
  it('renders columnChartGrouped component for snapshot test', () => {
    render(
      <ColumnChartGrouped
        id='columnChartGrouped'
        data={data}
        x={{ key: 'name' }}
        y={[
          {
            key: 'USA',
            start: 0,
          },
          { key: 'Europe' },
          { key: 'APAC' },
          { key: 'Africa' },
        ]}
      />
    );
    const columnChartGrouped = screen.getByTestId('columnChartGrouped');
    expect(columnChartGrouped).toMatchSnapshot();
  });
  it('renders columnChartGrouped component', () => {
    render(
      <ColumnChartGrouped
        id='columnChartGrouped'
        data={data}
        x={{ key: 'name' }}
        y={[
          {
            key: 'USA',
            start: 0,
          },
          { key: 'Europe' },
          { key: 'APAC' },
          { key: 'Africa' },
        ]}
      />
    );
    const columnChartGrouped = screen.getByTestId('columnChartGrouped');
    expect(columnChartGrouped).toBeInTheDocument();
  });
});
