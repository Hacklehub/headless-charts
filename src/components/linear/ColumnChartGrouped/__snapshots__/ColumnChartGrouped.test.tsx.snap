// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ColumnChartGrouped > renders columnChartGrouped component for snapshot test 1`] = `
<svg
  class="chart dark:bg-gray-800 dark:text-gray-50 h-64 lg:w-4/12 md:w-6/12 text-gray-900 w-full"
  data-testid="columnChartGrouped"
  id="columnChartGrouped"
>
  <g>
    <g>
      <rect
        class="fill-current undefined"
        height="-80"
        width="5.000000000000001"
        x="15.000000000000007"
        y="40"
      />
      <rect
        class="fill-current undefined"
        height="-75.19999999999999"
        width="5.000000000000001"
        x="-9.999999999999996"
        y="35.19999999999999"
      />
      <rect
        class="fill-current undefined"
        height="-64"
        width="5.000000000000001"
        x="-35"
        y="24"
      />
    </g>
    <g>
      <rect
        class="fill-current undefined"
        height="-32"
        width="5.000000000000001"
        x="20.000000000000007"
        y="-8"
      />
      <rect
        class="fill-current undefined"
        height="-72"
        width="5.000000000000001"
        x="-4.999999999999996"
        y="32"
      />
      <rect
        class="fill-current undefined"
        height="-32"
        width="5.000000000000001"
        x="-30"
        y="-8"
      />
    </g>
    <g>
      <rect
        class="fill-current undefined"
        height="-40"
        width="5.000000000000001"
        x="25.000000000000007"
        y="0"
      />
      <rect
        class="fill-current undefined"
        height="-32"
        width="5.000000000000001"
        x="5.329070518200751e-15"
        y="-8"
      />
      <rect
        class="fill-current undefined"
        height="-8"
        width="5.000000000000001"
        x="-25"
        y="-32"
      />
    </g>
    <g>
      <rect
        class="fill-current undefined"
        height="-8"
        width="5.000000000000001"
        x="30.00000000000001"
        y="-32"
      />
      <rect
        class="fill-current undefined"
        height="0"
        width="5.000000000000001"
        x="5.000000000000007"
        y="-40"
      />
      <rect
        class="fill-current undefined"
        height="0"
        width="5.000000000000001"
        x="-19.999999999999996"
        y="-40"
      />
    </g>
    <g
      class="yAxis axis"
      fill="none"
      font-family="sans-serif"
      font-size="10"
      text-anchor="end"
      transform="translate(40,0)"
    >
      <path
        class="domain"
        d="M-6,-39.5H0.5V40.5H-6"
        stroke="currentColor"
      />
      <g
        class="tick"
        opacity="1"
        transform="translate(0,-39.5)"
      >
        <line
          stroke="currentColor"
          x2="-6"
        />
        <text
          dy="0.32em"
          fill="currentColor"
          x="-9"
        >
          0
        </text>
      </g>
      <g
        class="tick"
        opacity="1"
        transform="translate(0,-31.5)"
      >
        <line
          stroke="currentColor"
          x2="-6"
        />
        <text
          dy="0.32em"
          fill="currentColor"
          x="-9"
        >
          1,000
        </text>
      </g>
      <g
        class="tick"
        opacity="1"
        transform="translate(0,-23.5)"
      >
        <line
          stroke="currentColor"
          x2="-6"
        />
        <text
          dy="0.32em"
          fill="currentColor"
          x="-9"
        >
          2,000
        </text>
      </g>
      <g
        class="tick"
        opacity="1"
        transform="translate(0,-15.5)"
      >
        <line
          stroke="currentColor"
          x2="-6"
        />
        <text
          dy="0.32em"
          fill="currentColor"
          x="-9"
        >
          3,000
        </text>
      </g>
      <g
        class="tick"
        opacity="1"
        transform="translate(0,-7.5)"
      >
        <line
          stroke="currentColor"
          x2="-6"
        />
        <text
          dy="0.32em"
          fill="currentColor"
          x="-9"
        >
          4,000
        </text>
      </g>
      <g
        class="tick"
        opacity="1"
        transform="translate(0,0.5)"
      >
        <line
          stroke="currentColor"
          x2="-6"
        />
        <text
          dy="0.32em"
          fill="currentColor"
          x="-9"
        >
          5,000
        </text>
      </g>
      <g
        class="tick"
        opacity="1"
        transform="translate(0,8.5)"
      >
        <line
          stroke="currentColor"
          x2="-6"
        />
        <text
          dy="0.32em"
          fill="currentColor"
          x="-9"
        >
          6,000
        </text>
      </g>
      <g
        class="tick"
        opacity="1"
        transform="translate(0,16.5)"
      >
        <line
          stroke="currentColor"
          x2="-6"
        />
        <text
          dy="0.32em"
          fill="currentColor"
          x="-9"
        >
          7,000
        </text>
      </g>
      <g
        class="tick"
        opacity="1"
        transform="translate(0,24.5)"
      >
        <line
          stroke="currentColor"
          x2="-6"
        />
        <text
          dy="0.32em"
          fill="currentColor"
          x="-9"
        >
          8,000
        </text>
      </g>
      <g
        class="tick"
        opacity="1"
        transform="translate(0,32.5)"
      >
        <line
          stroke="currentColor"
          x2="-6"
        />
        <text
          dy="0.32em"
          fill="currentColor"
          x="-9"
        >
          9,000
        </text>
      </g>
      <g
        class="tick"
        opacity="1"
        transform="translate(0,40.5)"
      >
        <line
          stroke="currentColor"
          x2="-6"
        />
        <text
          dy="0.32em"
          fill="currentColor"
          x="-9"
        >
          10,000
        </text>
      </g>
    </g>
    <g
      class="axis--x axis"
      fill="none"
      font-family="sans-serif"
      font-size="10"
      text-anchor="middle"
      transform="translate(0, -40)"
    >
      <path
        class="domain"
        d="M40.5,6V0.5H-39.5V6"
        stroke="currentColor"
      />
      <g
        class="tick"
        opacity="1"
        transform="translate(25.000000000000007,0)"
      >
        <line
          stroke="currentColor"
          y2="6"
        />
        <text
          dy="0.71em"
          fill="currentColor"
          y="9"
        >
          Product A
        </text>
      </g>
      <g
        class="tick"
        opacity="1"
        transform="translate(5.329070518200751e-15,0)"
      >
        <line
          stroke="currentColor"
          y2="6"
        />
        <text
          dy="0.71em"
          fill="currentColor"
          y="9"
        >
          Product B
        </text>
      </g>
      <g
        class="tick"
        opacity="1"
        transform="translate(-25,0)"
      >
        <line
          stroke="currentColor"
          y2="6"
        />
        <text
          dy="0.71em"
          fill="currentColor"
          y="9"
        >
          Product C
        </text>
      </g>
    </g>
  </g>
</svg>
`;
